daliuge-engine/test/manager/test_daemon.py

```diff
@@ -128,6 +128,79 @@ def test_zeroconf_discovery(self):
             "MasterManager didn't find the NodeManager running on the same node",
         )
 
+    def _test_zeroconf_dim_mm(self, disable_zeroconf=False):
+        # Start daemon with no master and no NM
+        self.create_daemon(master=False, noNM=True, disable_zeroconf=disable_zeroconf)
+        # Start DIM - now, on it's own
+        self._start("island", http.HTTPStatus.OK, {"nodes": []})
+        # Start daemon with master but no NM
+        self._start("master", http.HTTPStatus.OK)
+        # Check that dim registers to MM
+        timeout_time = time.time() + _TIMEOUT
+        dims = None
+        while time.time() < timeout_time:
+            dims = self._get_dims_from_master(_TIMEOUT)
+            if dims is not None and len(dims["islands"]) > 0:
+                break
+            time.sleep(0.1)
```

Comment:
This timeouting loop is repeated in a few places, can it be de-duplicated and reused?